.github/workflows/python.yml

```diff
@@ -23,45 +23,36 @@ permissions:
 
 jobs:
   linux:
-    runs-on: ${{ matrix.platform.runner }}
+    runs-on: ${{ matrix.runner }}
     strategy:
+      fail-fast: false
       matrix:
-        platform:
-          - runner: ubuntu-latest
-            target: x86_64
-          - runner: ubuntu-latest
-            target: x86
-          - runner: ubuntu-latest
-            target: aarch64
-          - runner: ubuntu-latest
-            target: armv7
-          - runner: ubuntu-latest
-            target: s390x
-          - runner: ubuntu-latest
-            target: ppc64le
+        runner: [ubuntu-latest]
+        target: [x86_64, x86, aarch64, armv7, s390x, ppc64le]
+        python-version: ["3.12"]
```

Comment:
DRYing - only testing against latest python, increasing the matrix size seems like over-kill

also I couldn't figure out how to refactor this block to be the same for the entire file

Reply:
> only testing against latest python, increasing the matrix size seems like over-kill

that's a bad idea because there are important differences between versions and we also had compatibility issues in the past because we didn't test

Reply:
I'm not against it and tried it in this PR with the 5 supported versions. It did slow down test running but I can certainly see the utility.

Reply:
ok, so the version on this line does not actually affect what versions we build/test. Actual images are built inside docker images

Reply:
From build log:
```
🐍 Found CPython 3.8, CPython 3.9, CPython 3.10, CPython 3.11, CPython 3.12, PyPy 3.8, PyPy 3.9, PyPy 3.10
```